### PR TITLE
Change default event loop for WIndows

### PR DESCRIPTION
**Pull request by sanders41**
@ThomasLaPiana can you see if this fixes this issue https://github.com/ethyca/fides/issues/513 ?

Note: I was unable to run the test suite even before making any changes because of a Snowflake error.

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@sanders41 i'll take a look, thank you!

**Comment by ThomasLaPiana** [comment]
@PSalant726 I'm assuming we're expecting this CI error due to hidden credentials?

**Comment by ThomasLaPiana** [comment]
ah, good point. will need to implement that fix over there too then and in fidesops, etc.

File: fideslog/sdk/python/client.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # pylint: disable=too-many-arguments
 
-from asyncio import run
+import asyncio
+import sys
+from asyncio import run, set_event_loop_policy
```
Comment (PSalant726) [inline fideslog/sdk/python/client.py:5]: `WindowsSelectorEventLoopPolicy` is not importable on macOS. Also, let's keep these as organized and minimal as possible.
```suggestion
from asyncio import SelectorEventLoop, run, set_event_loop
from selectors import SelectSelector
from sys import platform, version_info
```

File: fideslog/sdk/python/client.py [hunk 2]
```diff
@@ -60,6 +62,15 @@ def send(self, event: AnalyticsEvent) -> None:
...
         """
 
+        # There is a Python bug in the default Windows event loop for Python 3.8+. This works around
+        # the issue by changing the default event loop for Windows.
+        if (
+            sys.version_info[0] == 3
+            and sys.version_info[1] >= 8
+            and sys.platform.startswith("win")
+        ):
+            set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
```
Comment (PSalant726) [inline fideslog/sdk/python/client.py:72]:
```suggestion
		# Work around a Python bug in the default Windows event loop
		# by changing the default event loop for Windows.
		if (
			version_info[0] == 3  # Python v3.x.x
			and version_info[1] >= 8  # Python v3.8.x or greater
			and platform.lower().startswith("win")
		):
			set_event_loop(SelectorEventLoop(SelectSelector()))
```

**Review by PSalant726 — CHANGES_REQUESTED** [review]
@sanders41 Thanks for the PR, but I was getting `import-error`s while verifying things on macOS. LMK if my suggestions allow the issue to be resolved on all platforms.

@ThomasLaPiana
> I'm assuming we're expecting this CI error due to hidden credentials?

Correct, Paul's fork won't have access to the repo-level secrets in CI runs. I verified that all checks are passing locally (with my suggested changes).

File: fideslog/sdk/python/client.py (hunk 1 above)
Comment (ThomasLaPiana) [reply fideslog/sdk/python/client.py:5]: add `isort`?

File: fideslog/sdk/python/client.py (hunk 2 above)
Comment (ThomasLaPiana) [reply fideslog/sdk/python/client.py:72]: no bueno

**Comment by ThomasLaPiana** [comment]
can we not do a conditional import here and only import the windows selector if we detect the windows platform?

File: fideslog/sdk/python/client.py (hunk 1 above)
Comment (PSalant726) [inline fideslog/sdk/python/client.py:5]: `WindowsSelectorEventLoopPolicy` is not importable on macOS, so we'll need to import it after verifying that the process is running on Windows. Also, let's keep these as organized and minimal as possible.
```suggestion
from asyncio import run, set_event_loop_policy
from sys import platform, version_info
```

File: fideslog/sdk/python/client.py (hunk 2 above)
Comment (PSalant726) [inline fideslog/sdk/python/client.py:72]:
```suggestion
        # Work around a Python 3.8+ bug by changing
        # the default event loop for Windows.
        if (
            version_info[0] == 3  # Python v3.x.x
            and version_info[1] >= 8  # Python v3.8.x or greater
            and platform.lower().startswith("win")
        ):
            from asyncio import WindowsSelectorEventLoopPolicy  # type:ignore

            set_event_loop_policy(WindowsSelectorEventLoopPolicy())
```

**Review by PSalant726 — CHANGES_REQUESTED** [review]
> can we not do a conditional import here and only import the windows selector if we detect the windows platform? 

@ThomasLaPiana Sure. I was trying to avoid that, but give these suggestions a try instead.

File: fideslog/sdk/python/client.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # pylint: disable=too-many-arguments
```
Comment (PSalant726) [inline fideslog/sdk/python/client.py:1]: Also this gets rid of the pylint info-level message due to the conditional `import`:
```suggestion
# pylint: disable=import-outside-toplevel, too-many-arguments
```

**Review by PSalant726 — CHANGES_REQUESTED** [review]

File: fideslog/sdk/python/client.py (hunk 2 above)
Comment (ThomasLaPiana) [reply fideslog/sdk/python/client.py:72]: this looks good 👍

Comment (sanders41) [reply fideslog/sdk/python/client.py]: Since `set_event_loop_policy` is only needed if the os is Windows I moved that to the conditional imports also.

**Review by PSalant726 — APPROVED** [review]

**Pull request merged**